src/goto-symex/goto_state.cpp

```diff
@@ -12,6 +12,27 @@ Author: Romain Brenguier, romain.brenguier@diffblue.com
 
 #include <util/format_expr.h>
 
+optionalt<symbol_exprt>
+dereference_cachet::lookup(const exprt &dereference) const
+{
+  auto it = cache.find(dereference);
+  if(it == nullopt)
+  {
+    return nullopt;
+  }
+  else
+  {
+    return {it->get()};
+  }
```

Comment:
Wouldn't `return cache.find(dereference);` now do the trick?

Reply:
They are different types so sadly this does not work.

Reply:
Fixed in incoming commit.